src/mds/Server.cc

```diff
@@ -747,11 +753,21 @@ void Server::finish_flush_session(Session *session, version_t seq)
 }
 
 void Server::_session_logged(Session *session, uint64_t state_seq, bool open, version_t pv,
-			     interval_set<inodeno_t>& inos, version_t piv)
+			     interval_set<inodeno_t>& inos, version_t piv,
+			     interval_set<inodeno_t>& purge_inos, LogSegment *ls)
 {
-  dout(10) << "_session_logged " << session->info.inst << " state_seq " << state_seq << " " << (open ? "open":"close")
-	   << " " << pv << dendl;
+  dout(10) << "_session_logged " << session->info.inst
+	   << " state_seq " << state_seq
+	   << " " << (open ? "open":"close")
+	   << " " << pv
+	   << " purge_inos : " << purge_inos
+	   << " seq : " << ls->seq << dendl;
```

Comment:
Will segfault if `ls` is a nullptr. Testing of this PR found a segfault because `ls` could be garbage.

Reply:
updated it.

Reply:
Please also address my other comments.

Reply:
I'd like to hear from you about using semantics of move obj in constructors. This may have some side effects, I think.
thx.